core/io/generic.rs

```diff
@@ -32,6 +32,16 @@ pub struct GenericFile {
 }
 
 impl File for GenericFile {
+    // Since we let the OS handle the locking, file locking is not supported on the generic IO implementation
+    // No-op implementation allows compilation but provides no actual file locking.
```

Comment:
would it make sense to implement Drop trait to make sure that the `unlock_file` is called  when instance gets dropped?

Reply:
Agree

Reply:
@JoanFM @pereman2 I've added the Drop trait. Could you take another look when you get the chance? Thanks!